test-cypress/integration/explore/similarity.spec.js

```diff
@@ -0,0 +1,299 @@
+    };
+
+	let expandSimilarityIndexOptionsMenu = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        cy.get('div button[class="btn btn-link btn-sm mt-1 px-0 collapsed"]').should("be.visible");
+        cy.get('div button[class="btn btn-link btn-sm mt-1 px-0 collapsed"]').click();
+	};
+
+	let addBuildParam = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        expandSimilarityIndexOptionsMenu();
+        cy.get('#indexParameters').type(buildParam);
+        //Should add functionality to check whether the parameter is indeed passed to the field
+        cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click().wait(500);
+
+    };
+
+	let addStopWord = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        expandSimilarityIndexOptionsMenu();
+        cy.get('div[class="col-md-8"] input').eq(0).type(stopWord);
+        cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click().wait(500);
+
+    };
+
+	let addLuceneAnalyzer = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        expandSimilarityIndexOptionsMenu();
+        cy.get('div[class="col-md-8"] input').eq(1).type('{ctrl}a{backspace}', {force: true}).invoke('val', luceneAnalyzer).trigger('change', {force: true});
+        //cy.get('div[class="col-md-8"] input').eq(1).type(luceneAnalyzer);
+        cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click();
+    };
+
+	let checkLiteralIndex = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity/index/create');
+        expandSimilarityIndexOptionsMenu();
+        cy.get('div label[class="col-xs-6 col-md-6"] input').should('be.visible');
+        cy.get('div label[class="col-xs-6 col-md-6"] input').click().wait(500);
+        cy.get('button[class="btn btn-link btn-sm mt-1 px-0"]').click();
+    };
+
+	let createSimilarityIndex = function () {
+        cy.get('div button[class="btn btn-lg btn-primary"]').click();
+        cy.get('span i[class="icon-reload loader"]').should("be.visible");
+        cy.get('button span[class="icon-trash"]').should("be.visible");
+        cy.get('span i[class="icon-reload loader"]').should("not.be.visible");
+    };
+
+	let deleteSimilarityIndex = function () {
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+        cy.get('button span[class="icon-trash"]').should("be.visible");
+        cy.get('button span[class="icon-trash"]').click();
+        cy.get('button[class="btn btn-primary"]').click();
+        cy.get('div[class="card mb-2"]').should('be.visible').should('contain', 'No Indexes');
+    };
+
+	let selectIndexType = function (indexType) {
```

Comment:
Its safe to remove the `indexType` parameter and the check below as we don't pass anything else than `predication` string here. Later if we need such splitting, we might introduce it again. Also I suppose we could just generate the selector based on the condition instead of having such conditions e.g.
```javascript
let selectIndexType = function (indexType) {
  cy.get(`#create-${indexType}-index`).click();
}
```

Reply:
Removed the indexType parameter and the condition. Also renamed the function to switchToPredicationIndex.